bytes.go

```diff
@@ -17,10 +17,14 @@ func parseInt(bytes []byte) (v int64, ok bool, overflow bool) {
 	}
 
 	var n uint64 = 0
-	for _, c := range bytes {
+	for idx, c := range bytes {
 		if c < '0' || c > '9' {
 			return 0, false, false
 		}
+		// Deal with invalid data such as "00"
+		if c == '0' && n == 0 && idx != 0 {
```

Comment:
I believe this only restricts "00", but "01" isn't treated as an error. Can you expand this (and add another test)?

Reply:
ah good catch!